src/java/fr/paris/lutece/util/http/SecurityUtil.java

```diff
@@ -266,6 +271,75 @@ public static String getRealIp( HttpServletRequest request )
 
         return strIPAddress;
     }
+    
+    /**
+     * Validate a forward URL to avoid open redirect
+     * [see isRedirectUrlSafe(String, HttpServletRequest, String)]
+     * 
+     * @param strUrl
+     * @param request
+     * @return true if valid
+     */
+    public static boolean isRedirectUrlSafe( String strUrl, HttpServletRequest request )
+    {
+        return isRedirectUrlSafe( strUrl, request, null );
+    }
+
+
+    /**
+     * Validate a forward URL to avoid open redirect
+     * the url should :
+     *      - not be blank (null or empty string or spaces)
+     *      - not start with "http://" or "https://" or "//" OR match the base URL or any URL in the pattern list 
+     * 
+     * example with a base url "https://lutece.fr/ :
+     *      - valid : myapp/jsp/site/Portal.jsp , Another.jsp , https://lutece.fr/myapp/jsp/site/Portal.jsp
+     *      - invalid : http://anothersite.com , https://anothersite.com , //anothersite.com , file://my.txt , ...
+     * 
+     * 
+     * @param strUrl the Url to validate
+     * @param request the current request (containing the baseUrl)
+     * @param strAntPathMatcherPatterns a comma separated list of AntPathMatcher patterns, as "http://**.lutece.com,https://**.lutece.com"
+     * @return true if valid
+     */
+    public static boolean isRedirectUrlSafe( String strUrl, HttpServletRequest request, String strAntPathMatcherPatterns )
+    {
+
+        if ( StringUtils.isBlank( strUrl) ) return false ;
+	AntPathMatcher pathMatcher = new AntPathMatcher();

```

Comment:
inconsistent alignment
Plus you do not use that until the else clause. this should be moved there